pyproject.toml

```diff
@@ -20,18 +20,20 @@ dependencies = [
     "pydantic~=1.0",
     "loguru",
     "PyYaml>=6",
-    "typing_extensions>=3.10"
+    "typing-extensions>=3.10"
 ]
 
 [project.optional-dependencies]
 dev = [
-    "matplotlib>=3.5.3",
     "pytest>=7.1.2",
     "pytest-cov>=4.0.0",
     "pytest-codeblocks",
     "pre-commit",
     "multiprocessing_on_dill",
-    "pathos"
+    "pathos",
+    "igraph>=0.10.8",
+    "plotly>=5.18.0",
+    "numpy"
```

Comment:
please put these as extras... No need to install them all the time to use tawazi. 
And put the imports of numpy / p.otly / igraph inside of the used parts of the code

Reply:
you want to put them in optional deps different from dev ?

Reply:
fixed